src/Model/Item.php

```diff
@@ -1059,6 +1056,18 @@ private static function deleteById($item_id, $priority = PRIORITY_HIGH)
 
 		DBA::delete('item-delivery-data', ['iid' => $item['id']]);
 
+		if (!empty($item['iaid']) && !DBA::exists('item', ['iaid' => $item['iaid'], 'deleted' => false])) {
+			DBA::delete('item-activity', ['id' => $item['iaid']]);
+		}
+		if (!empty($item['icid']) && !DBA::exists('item', ['icid' => $item['icid'], 'deleted' => false])) {
+			DBA::delete('item-content', ['id' => $item['icid']]);
+		}
+		// When the permission set will be used in photo and events as well,
+		// this query here needs to be extended.
```

Comment:
I would love to see a multi-line comment block `/* bla */` here and not multiple single-line comments.